src/errors.rs

```diff
@@ -17,6 +17,11 @@ pub enum Error {
     /// The Meilisearch server returned an invalid JSON for a request.
     #[error("Error parsing response JSON: {}", .0)]
     ParseError(#[from] serde_json::Error),
+
+    /// An error occurred while parsing the fields of the response JSON.
+    #[error("Error parsing fields: {0}")]
+    ParseStringError(String),
+    
```

Comment:
Hey, I think it would be better if we didn't introduce any new error and instead returned hand-crafter `serde_json::Error` directly so our end user doesn't have to update their code to match this new error when it's, in fact, a deserialization error.

I’m ok with everything being a `custom_error`.
But if you really want, you can try to return more specific errors like`missing_field` and `invalid_value`

Reply:
But, if everything becomes a `custom error`, 

the code change will be huge.

Reply:
I have made the changes, requesting for a review

Reply:
Awesome thanks, sorry for the delay I was on holidays